satpy/dataset/metadata.py

```diff
@@ -27,20 +27,23 @@
 from satpy.writers.utils import flatten_dict
 
 
-def combine_metadata(*metadata_objects, average_times=True):
+def combine_metadata(*metadata_objects):
     """Combine the metadata of two or more Datasets.
 
     If the values corresponding to any keys are not equal or do not
     exist in all provided dictionaries then they are not included in
-    the returned dictionary.  By default any keys with the word 'time'
-    in them and consisting of datetime objects will be averaged. This
-    is to handle cases where data were observed at almost the same time
-    but not exactly.  In the interest of time, lazy arrays are compared by
-    object identity rather than by their contents.
+    the returned dictionary.  All values of the keys containing the substring
+    'start_time' will be set to the earliest value and similarly for 'end_time'
+    to latest time.  All other keys containing the word 'time' are averaged.
+    Before these adjustments, non-datetime objects are filtered out.
+
+    The same rules are applied to 'time_parameters' dictionary.
+
+    In the interest of processing time, lazy arrays are compared by object
+    identity rather than by their contents.
```

Comment:
What about reorganizing so all the time stuff is in the same paragraph:

```suggestion
    the returned dictionary.
    
    All values of the keys containing the substring
    'start_time' will be set to the earliest value and similarly for 'end_time'
    to latest time.  All other keys containing the word 'time' are averaged.
    Before these adjustments, non-datetime objects are filtered out.
    The same rules are applied to the 'time_parameters' sub-dictionary.

    In the interest of processing time, lazy arrays are compared by object
    identity rather than by their contents.
```

Otherwise, I wanted to ask what does it mean "non-datetime objects are filtered out"? Does that mean keys with "time" in them that are not datetime objects are removed and not averaged? Are they removed even if they're the same?

Reply:
Damn, didn't notice you had a suggestion here, already pushed my version. I also clarified the bit about "non-datetime objects"; it was meant mainly to say the `None` values coming from un-dated datasets were removed. And if there were any strings or such, they are also removed.